xcdat/temporal.py

```diff
@@ -863,7 +881,13 @@ def departures(
         inferred_freq = _infer_freq(ds[self.dim])
         if inferred_freq != freq:
             ds_obs = ds_obs.temporal.group_average(
-                data_var, freq, weighted, keep_weights, season_config, skipna
+                data_var,
+                freq,
+                weighted,
+                keep_weights,
+                season_config,
+                skipna,
+                min_weight,
```

Comment:
You set a `min_weight` in Steps 3 and 4. I wasn't sure if it was needed in Step 3, but I also couldn't think of problems of doing this. If you set `min_weight` to one and are missing one January of data, then all DJF values will be `NaN`'ed out, so it isn't critical that the first DJF value is `NaN`'ed out in Step 3 (but maybe there are different situations where it is important to apply a `min_weight` to both the original variable and the climatology.